packages/framework/fluid-static/src/rootDataObject.ts

```diff
@@ -214,24 +207,8 @@ class DOProviderContainerRuntimeFactory extends BaseContainerRuntimeFactory {
 			}
 			return entryPoint.get();
 		};
-		const getDefaultObject = async (
-			request: IRequest,
-			runtime: IContainerRuntime,
-			// eslint-disable-next-line unicorn/consistent-function-scoping
-		): Promise<IResponse | undefined> => {
-			const parser = RequestParser.create(request);
-			if (parser.pathParts.length === 0) {
-				// This cast is safe as ContainerRuntime.loadRuntime is called in the base class
-				return (runtime as ContainerRuntime).resolveHandle({
-					url: `/${rootDataStoreId}${parser.query}`,
-					headers: request.headers,
-				});
-			}
-			return undefined; // continue search
-		};
 		super({
 			registryEntries: [rootDataObjectFactory.registryEntry],
-			requestHandlers: [getDefaultObject],
```

Comment:
Want to sanity check this doesn't break LTS Loader support, as it seems this factory is only used internally for our various "Client" implementations.

From the bottom of [Removing-IFluidRouter.md](https://github.com/microsoft/FluidFramework/blob/main/packages/common/core-interfaces/Removing-IFluidRouter.md):

> The removal of some items will need to wait for the LTS version of the `Loader` to reach "2.0.0-internal.7.0.0". This is because old `Loader` or `Container` code doesn't know about the new `entryPoint` pattern and will still attempt to use the `request` pattern. The following items are affected:
> -   `requestHandler` property in `ContainerRuntime.loadRuntime(...)`, `BaseContainerRuntimeFactory`, `ContainerRuntimeFactoryWithDefaultDataStore`, `mixinAttributor`, `RuntimeFactory`, `TestContainerRuntimeFactory`

Reply:
Since AzureClient bundles it all together (loader + this factory), there shouldn't be a version mismatch to worry about.

However, I'm wondering how things are currently working for super old documents - I think these created their root data stores without aliasing (as I think this only got added later [here](https://github.com/microsoft/FluidFramework/commit/025ed3327d19b9ae24dc1710e7ec9945e3d7c918#diff-c40fee81fe1d30fba831dd5cf1189760d40e78b67cc7b8d42a25740a77271631R320)).  I think this would make those data stores only findable via request pattern.

So it seems that even without this change, we should be hitting `default dataStore [${rootDataStoreId}] must exist`?  I don't recall if this came up when the change from request to entrypoint was made [here](https://github.com/microsoft/FluidFramework/pull/18256/files#diff-92037cfad1a59cddfb8a5bca32b76f3d641899173c4ebc5116d3ba96004639e0R190), or if it's not actually a problem for some reason, or maybe I'm just forgetting something...?

Reply:
aliased datastores are backwards compatible with root datastores, so getAliasedDataStoreEntryPoint will get root datastores as well. as @ChumpChief is correct, since we bundle everything together this code path is never used, as the current service clients don't ever call request.